terraform/environments/electronic-monitoring-data/lambdas_iam.tf

```diff
@@ -409,3 +409,61 @@ resource "aws_iam_role_policy_attachment" "update_log_table_get_log_s3_files" {
     role = aws_iam_role.update_log_table.name
     policy_arn = aws_iam_policy.get_log_s3_files.arn
 }
+
+# ------------------------------------------
+# output_file_structure_as_json_from_zip
+# ------------------------------------------
+
+resource "aws_iam_role" "output_fs_json_lambda" {
+  name = "output_fs_json_lambda"
+  assume_role_policy  = data.aws_iam_policy_document.lambda_assume_role.json
+  managed_policy_arns = ["arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"]
+}
+
+data "aws_iam_policy_document" "output_fs_json_lambda_s3_policy_document" {
+  statement {
+    sid    = "S3Permissions"
+    effect = "Allow"
+    actions = [
+      "s3:GetObject",
+      "s3:PutObject",
+      "s3:ListBucket"
+    ]
+    resources = ["${aws_s3_bucket.data_store.arn}/*"]
+  }
+}
+
+resource "aws_iam_policy" "output_fs_json_lambda_s3_policy" {
+  name                = "output-fs-json-lambda-s3-policy"
+  description         = "Policy for Lambda to use S3 for ${local.output_fs_json_lambda}"
+  policy              = data.aws_iam_policy_document.output_fs_json_lambda_s3_policy_document.json
+}
+
+resource "aws_iam_role_policy" "output_fs_json_lambda_s3_policy" {
```

Comment:
do you need this role policy? i dont see it referenced anywhere else, and ive not had to use them previously in similar circumstances

Reply:
Hmmm perhaps not, probably left over from all of my IAM fiddling